packages/help-center/src/happychat-window-communicator.ts

```diff
@@ -47,8 +61,17 @@ export function useHCWindowCommunicator(
 							},
 							{ targetOrigin: event.origin }
 						);
-						// now clear the store, since we sent everything
-						resetStore();
+
+						break;
+					}
+					case 'happy-chat-authentication-data': {
+						event.source?.postMessage(
+							{
+								type: 'happy-chat-authentication-data',
+								authData: auth?.data,
```

Comment:
The optional chaining is not needed, `auth` is always an object (query result) and `auth.data` is always safe.

There is a race condition in this code: if the `useHappychatAuth` query hasn't finished fetching yet, this message handler will respond with `undefined`. Better to do something like:
```js
postMessage( {
  type: 'happy-chat-authentication-data',
  authData: queryClient.fetchQuery( 'getHappychatAuth' ),
} );
```

Reply:
The help-center doesn't have its own query client, it relies on the consumer to provide that. I'll use [`requestHappyChatAuth`](https://github.com/Automattic/wp-calypso/blob/trunk/packages/happychat-connection/src/use-happychat-auth.ts#L7) instead. It doesn't cache, but this happens once in most cases.

Reply:
Oh just realized that promises can't be cloned to be passed to another window, so we must pass the resolved data.

Reply:
> Oh just realized that promises can't be cloned

Oh, I wanted to write `await queryClient.fetchQuery()`, going to update the comment 🙂 

> The help-center doesn't have its own query client

If you can use `useQuery`, then you have a query client somewhere in React context. You can either use the `useQuery` hook, or access it directly with `const queryClient = useQueryClient()` and then do classic OOP-style programming on that object.